ls1intum/Artemis
# `Programming exercises`: Improve git-diff view on programming exercise detail page

### Checklist
#### General


#### Server
#### Client
#### Changes affecting Programming Exercises

### Motivation and Context


The current displaying of the git-diff on the programming exercise detail page is not very good looking. You have a button at the top and the rough stats at the bottom. It would make sense to combine them.

### Description

This PR moves the button next to the diff stats and modifies it to fit accordingly. While loading the diff, it shows a spinner and opens a modal afterward. 
The modal can now be closed by clicking outside of it.
A small bug that added a 10px margin to all buttons on the detail page instead of only to the button header at the top was fixed as well. This prevented proper centering of buttons next to text.

### Steps for Testing

Prerequisites:
- 1 Instructor/Editor/Tutor
- 1 Programming Exercise

1. Log in to Artemis
2. Navigate to the Programming Exercise Management Detail Page
3. Scroll down
4. See the described git-diff section
5. Hovering on the stats/button works
6. Clicking the button works as described

**Additional request:**
Please let me know if you have any suggestions for better placement of the diff section. I would like to move it further up but am unsure what the best place would be. It is a useful feature when creating and reviewing exercises, so I generally liked the quick access at the top.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage


Unchanged

### Screenshots
<img width="392" alt="grafik" src="https://user-images.githubusercontent.com/18218285/180101729-485e93c5-3ac8-4059-931d-91309282ced8.png">
<img width="392" alt="grafik" src="https://user-images.githubusercontent.com/18218285/180102271-7a3249ad-83aa-4011-b922-f90f8f1a80c7.png">
